cns/restserver/internalapi.go

```diff
@@ -245,12 +257,17 @@
 		logger.Errorf(returnMessage)
 	}
 
+	err = service.MarkExistingIPsAsPending(spec.IPsNotInUse)
```

Comment:
We dont need to call this during all createOrUpdateNC calls. This only needs to be marked one time during Reconcile. Can we move this to the caller API (ReconcileNCState)